docs/tls-authentication.md

```diff
@@ -0,0 +1,248 @@
+the `ca.conf` downloaded earlier:
+
+```
+[ server_alt_names ]
+DNS.1                   = localhost.localdomain
+DNS.2                   = localhost
+DNS.3                   = 127.0.0.1
+DNS.4                   = ::1
+```
+
+and replace DNS references with your actual domain. For instance:
+
+```
+[ server_alt_names ]
+DNS.1                   = www.example.com
+DNS.2                   = example.com
+```
+
+Then, follow the steps described in [Default | Running Script](#running-script) to generate the certificates.
+
+#### Different Expiry Date
+
+By default, certificates generated are valid for a period of 3 years. If you need a shorter (or
+longer) period, you can tweak the CA configuration to provide a different duration (in days).
+To do so, modify the `[ daedssrca ]` section in the `ca.conf` downloaded earlier:
+
+```
+[ daedssrca ]
+...
+default_days      = 1095
+default_crl_days  = 1095
+...
+```
+
+and replace the highlighted values above with whatever value you need. For instance, for a
+validity period of only one year:
+
+```
+[ daedssrca ]
+...
+default_days      = 365
+default_crl_days  = 365
+...
+```
+
+Then, follow the steps described in [Default | Running Script](#running-script) to generate the certificates.
+
+
+## Using Certificates
+
+### Setting Up Cardano-SL Backend
+
+Once the CA certificate, the server certificate and private key are in your possession, they
+can be fed to _cardano-sl_ as Command-Line Interface options:
+
+```
+$ cardano-node --help
+...
+Available options:
+...
+--tlscert FILEPATH       Path to file with TLS certificate
+--tlskey  FILEPATH       Path to file with TLS key
+--tlsca   FILEPATH       Path to file with TLS certificate authority
+...
+```
+
+For example:
+
+```
+$ cardano-node \
+  --tlscert ./tls/server/server.crt \
+  --tlskey  ./tls/server/server.key \
+  --tlsca   ./tls/ca/ca.crt 
+```
+
+All those options are actually optional. When missing, the node looks for default development
+certificates and key in `<repo-root>/scripts/tls-files/`. 
+
+#### Disable TLS (Not Recommended)
+
+If needed, you can disable TLS by providing the `--no-tls` flag to the wallet or by running a
+wallet in debug mode with `--wallet-debug` turned on.
+
+
+### Contacting Cardano-SL Backend
+
+When sending request to a node that has TLS enabled, clients need to provide their public
+certificate to the backend and successfuly perform a TLS handshake with it. This can be achieve
+in various ways depending on the client you use and the library available in your programming
+language. Make sure to provide the client certificate and key generatd in previous steps and
+signed by the same authority provided to the server.
+
+Additionally, to try things out on Unix systems, you may use `curl` as follows:
+
+```
+$ curl -kv --cert ./tls/client/client.crt --key ./tls/client/client.key <url>
+```
+
+The `k` flag ignores security checks on certificates (as they're done by the backend anyway),
```

Comment:
I'll admit, I use `-k` a lot for debugging purposes, but in documentation we give to the client, we really should recommend using `--cacert` to the ca certificate used to sign the server/client certificates instead, or better yet having them add the CA certificate to ca-bundle on any machines connecting to the API.

Reply:
You have a point. Will do :+1: